packages/Symfony/src/Rector/Class_/MakeCommandLazyRector.php

```diff
@@ -0,0 +1,155 @@
+{
+    public function configure()
+    {
+        $this->setName('sunshine');
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+use Symfony\Component\Console\Command\Command
+
+class SunshineCommand extends Command
+{
+    protected static $defaultName = 'sunshine';
+    public function configure()
+    {
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isType($node, self::COMMAND_CLASS)) {
+            return null;
+        }
+
+        $commandName = $this->resolveCommandNameAndRemove($node);
+        if ($commandName === null) {
+            return null;
+        }
+
+        $defaultNameProperty = $this->createDefaultNameProperty($commandName);
+        $node->stmts = array_merge([$defaultNameProperty], (array) $node->stmts);
+
+        return $node;
+    }
+
+    private function createDefaultNameProperty(Node $commandNameNode): Node\Stmt\Property
+    {
+        return $this->builderFactory->property('defaultName')
+            ->makeProtected()
+            ->makeStatic()
+            ->setDefault($commandNameNode)
+            ->getNode();
+    }
+
+    private function resolveCommandNameAndRemove(Class_ $class): ?Node
+    {
+        $commandName = null;
+        $this->callableNodeTraverser->traverseNodesWithCallable((array) $class->stmts, function (Node $node) use (
+            &$commandName
+        ) {
+            if ($node instanceof MethodCall) {
+                if (! $this->isName($node, 'setName')) {
+                    return null;
+                }
+
+                $commandName = $node->args[0]->value;
+                $this->removeNode($node);
+            }
+
+            if ($node instanceof ClassMethod && $this->isName($node, '__construct')) {
+                if (count((array) $node->stmts) !== 1) {
+                    return null;
+                }
+
+                $onlyNode = $node->stmts[0];
+                if ($onlyNode instanceof Expression) {
+                    $onlyNode = $onlyNode->expr;
+                    if (! $this->isName($onlyNode, '__construct')) {
+                        return null;
+                    }
+
+                    $commandName = $onlyNode->args[0]->value;
+                    if (! is_string($this->getValue($commandName))) {
+                        return null;
+                    }
+
+                    if (count($node->params) === 0) {
+                        $this->removeNode($node);
+                    }
+                }
+            }
+
+            if ($node instanceof StaticCall) {
+                if (! $this->isName($node, '__construct')) {
+                    return null;
+                }
+
+                $commandName = $node->args[0]->value;
```

Comment:
You need to check argument here too, cause I see:
```
+    protected static $defaultName = $entityManager;

-        parent::__construct($entityManager, $client);
+        parent::__construct($client);

Reply:
Also I see notice:
```
PHP Notice:  Trying to get property of non-object in /vendor/rector/rector/packages/Symfony/src/Rector/Class_/MakeCommandLazyRector.php on line 147
```
And error which may be related or not:
```
[ERROR] Could not process
         "/src/TestBundle/Command/SomeCommand.php" file, due to:
         "leaveNode() returned invalid value of type integer".
```

Reply:
What code is causing it? Error alone is not much helpful

Reply:
Was not able yet to separate it, will check later today or tomorrow.

Reply:
Seems like such command:
```php
class SomeCommand extends ContainerAwareCommand
{
    protected function configure()
    {
        $this
            ->setName('some:command')
            ->setDescription('Description')
            ->addArgument('id', InputArgument::REQUIRED, 'integer')
            ->addArgument('string', InputArgument::REQUIRED, 'string')
        ;
    }

    protected function execute(InputInterface $input, OutputInterface $output)
    {
        return true
    }
}
```
Is able to generate that issue.

Reply:
Perfect, thanks

Reply:
Full trace:
```
In NodeTraverser.php line 168:

  [LogicException]
  leaveNode() returned invalid value of type integer


Exception trace:
 () at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:168
 PhpParser\NodeTraverser->traverseNode() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:91
 PhpParser\NodeTraverser->traverse() at /vendor/rector/rector/src/PhpParser/Node/Commander/NodeRemovingCommander.php:53
 Rector\PhpParser\Node\Commander\NodeRemovingCommander->traverseNodes() at /vendor/rector/rector/src/Rector/AbstractRector.php:126
 Rector\Rector\AbstractRector->afterTraverse() at /vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:94
 PhpParser\NodeTraverser->traverse() at /vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php:52
 Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverse() at /vendor/rector/rector/src/Application/FileProcessor.php:114
 Rector\Application\FileProcessor->refactor() at /vendor/rector/rector/src/Application/RectorApplication.php:108
 Rector\Application\RectorApplication->Rector\Application\{closure}() at /vendor/rector/rector/src/Application/RectorApplication.php:137
 Rector\Application\RectorApplication->tryCatchWrapper() at /vendor/rector/rector/src/Application/RectorApplication.php:109
 Rector\Application\RectorApplication->runOnFileInfos() at /vendor/rector/rector/src/Console/Command/ProcessCommand.php:154
 Rector\Console\Command\ProcessCommand->execute() at /vendor/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /vendor/rector/rector/src/Console/Command/AbstractCommand.php:32
 Rector\Console\Command\AbstractCommand->run() at /vendor/symfony/console/Application.php:926
 Symfony\Component\Console\Application->doRunCommand() at /vendor/symfony/console/Application.php:269
 Symfony\Component\Console\Application->doRun() at /vendor/rector/rector/src/Console/Application.php:92
 Rector\Console\Application->doRun() at /vendor/symfony/console/Application.php:145
 Symfony\Component\Console\Application->run() at /vendor/rector/rector/bin/rector:24

Reply:
Stack trace is very rarely needed. The best is to provide:

- src file that causes the error (the smaller the better, it's usually 1 line)
- error message
- Rector class that causes the error